### PR TITLE
sqs: service docs

### PR DESCRIPTION
**Pull request by chrisfarms**
## What

adds docs for using the SQS service

* creating a service, difference between "fifo" and "standard" (and polling for ready)
* updating a service, what options you have to configure it
* how to set up your secondary queue as a dead-letter queue
* binding (and polling for ready)
* unbinding (and polling for ready)
* destroying

### CODE REVIEW

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [inline source/documentation/deploying_services/sqs.md:159]: Should we remove this "connect from outside" section for SQS for now? 

... I'm tempted to not document this yet as this feels like a bug not a feature that you can connect to the queue from outside the PaaS like this... we should make a decision on what default behaviour we want before documenting it

Comment (Krenair) [reply source/documentation/deploying_services/sqs.md:159]: I'm not sure whether it's either, I recall the S3 broker had some of the plumbing restricting this (which we didn't keep). I would suggest not documenting it for now.

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [inline source/documentation/deploying_services/sqs.md:155]: need a better word than "isolated"

Comment (Krenair) [reply source/documentation/deploying_services/sqs.md:155]: separate?

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:3]:
```suggestion
Amazon SQS provides "Standard" (at-least-once delivery, best-effort ordering) and First-In-First-Out queues.
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:27]:
```suggestion
queue, letting you isolate problematic messages to determine why their
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:28]:
```suggestion
processing did not succeed.
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:23]:
```suggestion
consumer processes it and deletes it.
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:42]:
```suggestion
It will take between 1 and 5 minutes to set up the service instance. To check
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:85]:
```suggestion
It will take between 1 and 5 minutes to set up the binding. To check its progress, run:
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:108]:
```suggestion
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:110]:
```suggestion
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:150]: I think this wording makes it sound like using the right AWS SDK will make the SDK know how to deal with VCAP_SERVICES

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:251]:
```suggestion
a period in seconds to retain messages for. This parameter can also be passed at service creation time.
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:278]:
```suggestion
- `secondary_queue_url` is the URL for the secondary queue, now acting as a dead-letter queue
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:288]:
```suggestion
It can take between 1 and 5 minutes to delete the service instance. To check
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [reply source/documentation/deploying_services/sqs.md:155]:
```suggestion
Your app may also connect to a second, separate, queue using the `secondary_queue_url`
```

File: source/documentation/deploying_services/sqs.md
Comment (Krenair) [inline source/documentation/deploying_services/sqs.md:273]:
```suggestion
redirected to the secondary queue. This parameter can also be passed at service creation time.
```

**Review by Krenair — APPROVED** [review]

**Comment by Krenair** [comment]
@chrisfarms I contributed to this but you're the PR author, I've left an approval but you should probably merge it to signify approval of my changes

**Comment by chrisfarms** [comment]
We have said we will not merge this to the public docs until we are ready to release into general availability and it will require tech writer approval etc. As such, I'm going to put it back into draft state for now.

**Comment by Krenair** [comment]
Oh, so we did, thanks

>

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [inline source/documentation/deploying_services/sqs.md:109]:
```suggestion
- `consumer` is appropriate for apps which will be reading and possibly deleting messages,
```

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [inline source/documentation/deploying_services/sqs.md:25]: Think we should link to the section below about configuring dead-letter queue from here somewhere

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:3]: Would be good to provide links for all these concepts

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:16]: Need a comma before 'named'

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:18]: Bullets should follow on from the lead-in sentence, so we should probably not use bullets here

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:26]: We should add a link for 'dead-letter queue'

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:54]: We should use 'cannot' rather than 'can't'

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:62]: You have a typo in 'minutes'

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:106]: these bullet points need to follow on grammatically from the lead-in sentence, so need some tweaking

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:110]: Bullet points should not have full stops :)

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:164]: Should these be in code blocks like VCAP_SERVICES is?

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:172]: Link seems broken

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:16]:
```suggestion
The name of the service overall is `aws-sqs-queue`. There are two types of plan
available for the service, `standard` and `fifo`.
```

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:54]:
```suggestion
is critical, or where duplicates cannot be tolerated.
```

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:62]:
```suggestion
It will take between 1 and 5 minutes to set up the service instance. To check
```

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:110]:
```suggestion
The currently supported `access_policy` values are:

- `full`, which means the default, full access permissions
- `producer`, which is appropriate for an app that will be sending messages to the queue,
  but does not need to be able to read or delete messages
- `consumer`, which is appropriate for an app that will be reading and possibly deleting messages,
  but does not need to be able to send messages to the queue
```

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:3]: These aren't really concepts that have much follow up information, however I have removed them from this section so now they only appear in the section that links off to the AWS docs, so that can hopefully serve for more of a deep dive into the different types.

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:18]: I have kept the bullets but tweaked this paragraph so that it better conforms to the standard

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:26]: done

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:164]: yes they should...done

File: source/documentation/deploying_services/sqs.md
Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:172]: yep ... fixed

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [reply source/documentation/deploying_services/sqs.md]: yes, this is better

**Review by cmcnallygds — APPROVED** [review]

**Review by 46bit — APPROVED** [review]
I think parts of this are in language that could be simplified, but it's already pretty understandable and good to ship.

**Comment by 46bit** [comment]
⚠️ SQS is not enabled for our tenants yet, because of questions around billing. Don't merge this as it'll be documenting something we don't support yet ⚠️

**Comment by LeePorte** [comment]
We should add a note to say that it is a beta service in testing and if you would like to test the service get in contact with support. That way we get some feedback and we can announce it to our tenants.

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:6]:
```suggestion
This is a beta service which is currently in a testing phase. If you would like to test the service, contact the GOV.UK PaaS team at [gov-uk-paas-support@digital.cabinet-office.gov.uk](mailto:gov-uk-paas-support@digital.cabinet-office.gov.uk). 
```

Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:6]: Do we still need this line about feedback, or is the one about testing the service supposed to replace it?

Comment (chrisfarms) [reply source/documentation/deploying_services/sqs.md:6]: I don't think we need both

File: source/documentation/deploying_services/sqs.md
Comment (cmcnallygds) [inline source/documentation/deploying_services/sqs.md:7]:
```suggestion

```

**Pull request merged**
